geopandas/plotting.py

```diff
@@ -119,16 +134,11 @@ def plot_polygon_collection(
     if "markersize" in kwargs:
         del kwargs["markersize"]
 
+    # Add to kwargs for easier checking below.
     if color is not None:
         kwargs["color"] = color
-        _check_and_expand(kwargs, "color", multiindex, is_color_like)
-    else:
-        for att in ["facecolor", "edgecolor"]:
-            if att in kwargs:
-                _check_and_expand(kwargs, att, multiindex, is_color_like)
```

Comment:
I think this was a bit inconsistent. On the one hand, the value for `facecolor`/`edgecolor` was only checked and corrected if `color` was not specified. Yet on the other hand, the keyword was passed on to the collection regardless.
My proposed solution is to let this function concentrate on expanding the values and let the caller and `mpl` deal with any overlappingly specified properties. No need to reinvent the wheel